src/crypto/secrets.ts

```diff
@@ -40,3 +48,35 @@ export const saveAccountSecrets = async (initialiseWallet: SerializedWallet) =>
 
   return savePrivateKey(uuid, privateKey);
 };
+
+/**
+ * Gets and decrypts the settings key if it exists. Otherwise, a new settings key is generated and
+ * stored in the keychain.
+ *
+ * This assumes that the global `encryptionKey` is set.
+ */
+export const getSettingsKey = async (): Promise<Buffer> => {
+  const settingsKey = await getPrivateKey(KEYTAR_SETTINGS_KEY_NAME);
+  if (settingsKey) {
+    return Buffer.from(settingsKey, 'hex');
+  }
+
+  const newKey = createEncryptionKey();
+  await savePrivateKey(KEYTAR_SETTINGS_KEY_NAME, newKey.toString('hex'));
+
+  return newKey;
+};
+
+export const hasSettingsKey = async (): Promise<boolean> => {
+  const key = await keytar.getPassword(KEYTAR_SERVICE, KEYTAR_SETTINGS_KEY_NAME);
+  return !!key;
+};
+
+export const checkSettingsKey = async (): Promise<boolean> => {
+  try {
+    await getSettingsKey();
```

Comment:
I'm worried this will potentially cause a weird bug where we accidentally create a settings key when we don't really want to and possibly have a false positive.